Fixes needed to make login with the Keycard work

saledjenic opened the pull request.

saledjenic committed:
fix(keycard): subaccounts updated wiht keyuid, colorhash and colorid

saledjenic committed:
feat: `GetDerivedAddressDetails` added to accounts api

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (15)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | bffbacfd | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2916/1/) | 2022-10-21 15:07:54 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221021-150445-bffbac-pr2916.zip) |
| :x: | bffbacfd | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/1/) | 2022-10-21 15:08:06 | ~3 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/1/consoleText) |
| :heavy_check_mark: | bffbacfd | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2916/1/) | 2022-10-21 15:10:21 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221021-150445-bffbac-pr2916.aar) |
| | | | | | | |
| :heavy_check_mark: | 311366c4 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2916/2/) | 2022-10-24 08:00:34 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221024-075801-311366-pr2916.zip) |
| :x: | 311366c4 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/2/) | 2022-10-24 08:01:35 | ~3 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/2/consoleText) |
| :heavy_check_mark: | 311366c4 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2916/2/) | 2022-10-24 08:03:02 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221024-075801-311366-pr2916.aar) |
| | | | | | | |
| :heavy_check_mark: | d5c6b417 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2916/3/) | 2022-10-26 17:29:43 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221026-172725-d5c6b4-pr2916.zip) |
| :heavy_check_mark: | d5c6b417 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/3/) | 2022-10-26 17:30:21 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-221026-172720-d5c6b4-pr2916.zip) |
| :heavy_check_mark: | d5c6b417 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2916/3/) | 2022-10-26 17:32:08 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221026-172720-d5c6b4-pr2916.aar) |
| | | | | | | |
| :x: | 6b5f9943 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/4/) | 2022-10-26 20:07:11 | ~57 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/4/consoleText) |
| :heavy_check_mark: | 6b5f9943 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2916/4/) | 2022-10-26 20:08:15 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221026-200613-6b5f99-pr2916.zip) |
| :heavy_check_mark: | 6b5f9943 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2916/4/) | 2022-10-26 20:09:49 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221026-200613-6b5f99-pr2916.aar) |
| | | | | | | |
| :x: | 44e94430 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/5/) | 2022-10-27 08:13:32 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/5/consoleText) |
| :heavy_check_mark: | 44e94430 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2916/5/) | 2022-10-27 08:15:37 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221027-081221-44e944-pr2916.zip) |
| :heavy_check_mark: | 44e94430 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2916/5/) | 2022-10-27 08:17:55 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221027-081221-44e944-pr2916.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 94204a19 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2916/6/) | 2022-10-28 07:21:58 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221028-071920-94204a-pr2916.zip) |
| :heavy_check_mark: | 94204a19 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/6/) | 2022-10-28 07:22:01 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-221028-071920-94204a-pr2916.zip) |
| :heavy_check_mark: | 94204a19 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2916/6/) | 2022-10-28 07:24:59 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221028-071920-94204a-pr2916.aar) |
| | | | | | | |
| :heavy_check_mark: | a8d66cc9 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2916/7/) | 2022-10-28 11:12:39 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221028-111041-a8d66c-pr2916.zip) |
| :heavy_check_mark: | a8d66cc9 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2916/7/) | 2022-10-28 11:13:12 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-221028-111041-a8d66c-pr2916.zip) |
| :heavy_check_mark: | a8d66cc9 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2916/7/) | 2022-10-28 11:14:41 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221028-111037-a8d66c-pr2916.aar) |

saledjenic committed:
feat: 5 functions added, the same as we had, but without pass verification (need for a keycard users)

The following three new functions introduced, for which password should be verified
on the client side (in case of a keycard user we don't have keystores to check pass):
- `AddAccountWithMnemonicPasswordVerified`
- `AddAccountWithMnemonicAndPathPasswordVerified`
- `AddAccountWithPrivateKeyPasswordVerified`
- `GenerateAccountPasswordVerified`
- `GenerateAccountWithDerivedPathPasswordVerified`

update

Samyoul commented on services/accounts/accounts.go:
```diff
@@ -221,7 +301,7 @@ func (api *API) verifyPassword(password string) error {
 }
 
-func (api *API) addAccountWithMnemonic(
+func (api *API) addAccountWithMnemonicPasswordVerified(
```
Does the name `addAccountWithMnemonicPasswordVerified` mean that the password has already been verified?

If so, perhaps you could add some go docs to explain the function. Something like:

```suggestion
// addAccountWithMnemonicPasswordVerified adds an accounts.Account derived from the given Mnemonic ... (some info about the usage of the password)
func (api *API) addAccountWithMnemonicPasswordVerified(
```

saledjenic replied:
@Samyoul thanks, yes, will do that.

Samyoul reviewed (approved):
I don't see any problem with the implementation looks good. My only question is make the function understandable for future developers, and adding some go docs to the function would greatly help with that.
